packages/admin/src/table/filterbar/FilterBarActiveFilterBadge.tsx

```diff
@@ -0,0 +1,72 @@
+import { createStyles, Theme, Typography, WithStyles, withStyles } from "@material-ui/core";
+import * as React from "react";
+import { useForm } from "react-final-form";
+
+export type CometAdminFilterBarActiveFilterBadgeClassKeys = "hasValueCount";
+
+export const dirtyFieldsCount = (values: Record<string, any>, registeredFields: string[] = []) => {
+    let count = 0;
+    Object.entries(values).forEach(([fieldName, fieldValue]) => {
+        if (!fieldValue) {
+            return;
+        }
+
+        const isRegisteredField = registeredFields.includes(fieldName);
+        if (Array.isArray(fieldValue) && isRegisteredField) {
+            count += fieldValue.length;
+        } else if (typeof fieldValue === "object") {
+            if (Object.keys(fieldValue).length === 2 && "min" in fieldValue && "max" in fieldValue && isRegisteredField) {
+                count++;
+            } else {
+                count += dirtyFieldsCount(fieldValue, registeredFields);
+            }
+        } else if (isRegisteredField) {
+            count++;
+        }
+    });
+
+    return count;
+};
+
+const styles = (theme: Theme) =>
+    createStyles<CometAdminFilterBarActiveFilterBadgeClassKeys, FilterBarActiveFilterBadgeProps>({
+        hasValueCount: {
+            backgroundColor: `${theme.palette.grey[100]}`,
+            textAlign: "center",
+            borderRadius: "4px",
+            padding: "0 5px",
+            height: "20px",
+        },
+    });
+
+export interface FilterBarActiveFilterBadgeProps {
+    values: Record<string, any>;
+    calcNumberDirtyFields?: (values: Record<string, any>, registeredFields: string[]) => number;
+}
+
+export const FilterBarActiveFilterBadgeComponent: React.FC<WithStyles<typeof styles, true> & FilterBarActiveFilterBadgeProps> = ({
+    values,
+    calcNumberDirtyFields = dirtyFieldsCount,
+    classes,
+}) => {
+    const form = useForm();
+    const registeredFields = form.getRegisteredFields().map((field) => field.split(".").pop() as string);
+
+    const countValue = React.useMemo(() => {
+        return calcNumberDirtyFields(values, registeredFields);
+    }, [values, registeredFields, calcNumberDirtyFields]);
```

Comment:
why this useMemo? I think it won't have any effect as registeredFields gets re-created on every render (due to the map you do)

Reply:
thats true i thought i would need a state :) and the useMemo approach showed me Obi. i had a useState and a useEffect before :) but i removed it and it still works so thanks for the info :)